qcodes/instrument_drivers/stanford_research/SR830.py

```diff
@@ -1,5 +1,103 @@
+from functools import partial
+import numpy as np
+
 from qcodes import VisaInstrument
-from qcodes.utils.validators import Numbers, Ints, Enum, MultiType
+from qcodes.instrument.parameter import ArrayParameter
+from qcodes.utils.validators import Numbers, Ints, Enum, Strings
+
+
+class ChannelBuffer(ArrayParameter):
+    """
+    Parameter class for the two channel buffers
+
+    Currently always returns the entire buffer
+    TODO (WilliamHPNielsen): Make it possible to query parts of the buffer.
+        The instrument natively supports this in its TRCL call.
+    """
+
+    def __init__(self, name, instrument, channel):
+        """
+        Args:
+            name (str): The name of the parameter
+            instrument (SR830): The parent instrument
+            channel (int): The relevant channel (1 or 2). The name should
+                should match this.
+        """
+        if channel not in [1, 2]:
+            raise ValueError('Invalid channel specifier. SR830 only has '
+                             'channels 1 and 2.')
+
+        if not isinstance(instrument, SR830):
+            raise ValueError('Invalid parent instrument. ChannelBuffer '
+                             'can only live on an SR830.')
+
+        super().__init__(name,
+                         shape=(1,),  # dummy initial shape
+                         unit='V',  # dummy initial unit
+                         setpoint_names=('Time',),
+                         setpoint_labels=('Time',),
+                         setpoint_units=('s',),
+                         docstring='Holds an acquired (part of the) '
+                                   'data buffer of one channel.')
+
+        self.channel = channel
+        self._instrument = instrument
+
+    def prepare_buffer_readout(self):
+        """
+        Function to generate the setpoints for the channel buffer and
+        get the right units
+        """
+
+        N = self._instrument.buffer_npts()  # problem if this is zero?
+        # TODO (WilliamHPNielsen): what if SR was changed during acquisition?
+        dt = 1/self._instrument.buffer_SR()
+        self.shape = (N,)
+        self.setpoints = (tuple(np.linspace(0, N*dt, N)),)
+
+        params = self._instrument.parameters
+        # YES, it should be: "is not 'none'" NOT "is not None"
```

Comment:
`'none'` is what the manual calls it, but I'm fine with `'off'` or even `'identity'`.